go/vt/wrangler/vexec_plan.go

```diff
@@ -17,48 +17,186 @@ limitations under the License.
+	"context"
 	"fmt"
+	"strings"
 
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
 	"vitess.io/vitess/go/vt/sqlparser"
+
+	"github.com/olekukonko/tablewriter"
 )
 
+// vexecPlan contains the final query to be sent to the tablets
 type vexecPlan struct {
-	query       string
 	opcode      int
 	parsedQuery *sqlparser.ParsedQuery
 }
 
+// vexecPlannerParams controls how some queries/columns are handled
+type vexecPlannerParams struct {
+	dbNameColumn         string
+	workflowColumn       string
+	immutableColumnNames []string
+	updatableColumnNames []string
+	updateTemplates      []string
+}
+
+// vexecPlanner generates and executes a plan
+type vexecPlanner interface {
+	params() *vexecPlannerParams
+	exec(ctx context.Context, masterAlias *topodatapb.TabletAlias, query string) (*querypb.QueryResult, error)
+	dryRun() error
+}
+
+// vreplicationPlanner is a vexecPlanner implementation, specific to _vt.vreplication table
+type vreplicationPlanner struct {
+	vx *vexec
+	d  *vexecPlannerParams
+}
+
+func newVreplicationPlanner(vx *vexec) vexecPlanner {
+	return &vreplicationPlanner{
+		vx: vx,
+		d: &vexecPlannerParams{
+			dbNameColumn:         "db_name",
+			workflowColumn:       "workflow",
+			immutableColumnNames: []string{"id"},
+			updatableColumnNames: []string{},
+		},
+	}
+}
+func (p vreplicationPlanner) params() *vexecPlannerParams { return p.d }
+func (p vreplicationPlanner) exec(
+	ctx context.Context, masterAlias *topodatapb.TabletAlias, query string,
+) (*querypb.QueryResult, error) {
+	qr, err := p.vx.wr.VReplicationExec(ctx, masterAlias, query)
+	if err != nil {
+		return nil, err
+	}
+	if qr.RowsAffected == 0 {
+		return nil, fmt.Errorf("\nno matching streams found for workflow %s, tablet %s, query %s", p.vx.workflow, masterAlias, query)
+	}
+	return qr, nil
+}
+func (p vreplicationPlanner) dryRun() error {
+	rsr, err := p.vx.wr.getStreams(p.vx.ctx, p.vx.workflow, p.vx.keyspace)
+	if err != nil {
+		return err
+	}
+
+	p.vx.wr.Logger().Printf("Query: %s\nwill be run on the following streams in keyspace %s for workflow %s:\n\n",
+		p.vx.plannedQuery, p.vx.keyspace, p.vx.workflow)
+	tableString := &strings.Builder{}
+	table := tablewriter.NewWriter(tableString)
+	table.SetHeader([]string{"Tablet", "ID", "BinLogSource", "State", "DBName", "Current GTID", "MaxReplicationLag"})
+	for _, master := range p.vx.masters {
+		key := fmt.Sprintf("%s/%s", master.Shard, master.AliasString())
+		for _, stream := range rsr.Statuses[key] {
+			table.Append([]string{key, fmt.Sprintf("%d", stream.ID), stream.Bls.String(), stream.State, stream.DBName, stream.Pos, fmt.Sprintf("%d", stream.MaxReplicationLag)})
+		}
+	}
+	table.SetAutoMergeCellsByColumnIndex([]int{0})
+	table.SetRowLine(true)
+	table.Render()
+	p.vx.wr.Logger().Printf(tableString.String())
+	p.vx.wr.Logger().Printf("\n\n")
+
+	return nil
+}
+
+// vreplicationPlanner is a vexecPlanner implementation, specific to _vt.schema_migrations table
```

Comment:
should be 
// schemaMigrationsPlanner is a ...

Reply:
fixed!